process/preprocess/eventsPreProcessorV0.go

```diff
@@ -72,6 +67,53 @@ func (d *eventsPreProcessorV0) SaveBlock(marshalledData []byte) error {
 	return nil
 }
 
+func (d *eventsPreProcessorV0) setTransactionsPool(txsPool *data.TransactionsPool) (*outport.TransactionPool, error) {
+	if txsPool == nil {
+		return nil, process.ErrNilTransactionsPool
+	}
+
+	txs := make(map[string]*outport.TxInfo)
+
+	if txsPool.Txs != nil {
+		for hash, txHandler := range txsPool.Txs {
+			txs[hash] = &outport.TxInfo{
+				Transaction:    txHandler.TransactionHandler,
```

Comment:
as txHandler is a pointer, I think we should check for nil before this line in order to avoid panics.. same for scrHandler on L92 and logHandler on L103

Reply:
added more nil checks